### PR TITLE
Finding numbers. With threads and without one.

### PR DESCRIPTION
**Pull request by Gaddsky**
In this variant I have not find simplest right way to measure time of programm execution with threads. All methods I found are very inexplicit for me.
Results are:

Without threads starts
3816547290
Without threads ends in 196344ms

With 5 threads starts in 1459629493749
3816547290
Stop time 1459629546944
Stop time 1459629546993
Stop time 1459629547125
Stop time 1459629547291
Stop time 1459629547674

(It means 53925 ms with 5 threads)

Sorry for my English.

### CODE REVIEW

File: gadd/FindNumber/Run.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+        int numberOfThreads = 5;
+        startTime = System.currentTimeMillis();
+        System.out.println("\nWith " + numberOfThreads + " threads starts in " + startTime);
+        long start = 123456789L, stop = 987654321L;
+        long step = (stop - start) / numberOfThreads;
+        Thread[] threads = new Thread[numberOfThreads];
+        for (int i = 0; i < numberOfThreads; i++) {
+            Runnable run = new NumberFinderThreads(start * 10 + step * 10 * i, start * 10 + step * 10 * (i + 1));
+            threads[i] = new Thread(run);
```
Comment (pasby) [inline gadd/FindNumber/Run.java]: you can start threads right away, not need to store them

File: gadd/FindNumber/NumberFinderNoThreads.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
+package gadd.FindNumber;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+
+
+public class NumberFinderNoThreads {
```
Comment (pasby) [inline gadd/FindNumber/NumberFinderNoThreads.java]: you need add 'implements Runnable' here

File: gadd/FindNumber/NumberFinderNoThreads.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        }
+        return true;
+    }
+
+
+    protected void go(long startNumber, long stopNumber) {
+        for (long i = startNumber; i <= stopNumber; i += 10) {
+            if (checkDifferentDigits(i)) {
+                if (checkRule(i)) {
+                    System.out.println(i);
```
Comment (pasby) [inline gadd/FindNumber/NumberFinderNoThreads.java]: better make it return result

File: gadd/FindNumber/Run.java [hunk 4]
```diff
@@ -0,0 +1,27 @@
+package gadd.FindNumber;
+
+public class Run {
```
Comment (pasby) [inline gadd/FindNumber/Run.java]: you can add public syncronized flag to check whether result if found

File: gadd/FindNumber/NumberFinderThreads.java [hunk 5]
```diff
@@ -0,0 +1,14 @@
+package gadd.FindNumber;
+
+public class NumberFinderThreads extends NumberFinderNoThreads implements Runnable {
```
Comment (pasby) [inline gadd/FindNumber/NumberFinderThreads.java]: I like this thing, it is good to reuse code if it is possible

File: gadd/FindNumber/NumberFinderNoThreads.java [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+
+
+public class NumberFinderNoThreads {
+//    private long start = 1234567890L, stop = 9876543210L;
+
+    public NumberFinderNoThreads() {
```
Comment (pasby) [inline gadd/FindNumber/NumberFinderNoThreads.java]: don't need this constructor

File: gadd/FindNumber/NumberFinderNoThreads.java [hunk 7]
```diff
@@ -0,0 +1,50 @@
+package gadd.FindNumber;
```
Comment (pasby) [inline gadd/FindNumber/NumberFinderNoThreads.java]: packages should started from small letter 'findnumber'

File: gadd/FindNumber/Run.java (hunk 1 above)
Comment (Gaddsky) [inline gadd/FindNumber/Run.java]: Initially it was as you say. My mistake, I forgot change it back.

**Pull request merged**